bc_obps/compliance/service/supplementary_version_service.py

```diff
@@ -43,16 +44,23 @@ def handle(
     ) -> Optional[ComplianceReportVersion]: ...
 
 
-# Concrete strategy for flagging compliance report versions as requiring manual handling and for assigning 0 excess and credited emissions when the previous version also required manual handling. (There are too many variables in how a version can be manually handled in ways that the app is not aware of, so we can't really safely and accurately predict how to handle any subsequent child versions once a version has been closed via actions outside of the app.)
+# Concrete strategy for:
+# - Detecting when a previous CRV is still under manual handling (pending), and
+# - Creating a new supplementary NO_OBLIG / earned-credits CRV
+#   with an associated manual-handling record to carry that flag forward.
 class ManualHandler:
     @staticmethod
-    def can_handle(new_summary: ReportComplianceSummary, previous_summary: ReportComplianceSummary) -> bool:
-        previous_compliance_report_version = ComplianceReportVersion.objects.get(
-            report_compliance_summary=previous_summary
-        )
-        if previous_compliance_report_version.requires_manual_handling:
-            return True
-        return False
+    def can_handle(
+        new_summary: ReportComplianceSummary,
+        previous_summary: ReportComplianceSummary,
+    ) -> bool:
+        """
+        This handler applies when the *previous* compliance report version has a manual-handling record
+        """
+        previous_crv = ComplianceReportVersion.objects.get(report_compliance_summary=previous_summary)
+
+        # Use the existence of the one-to-one manual_handling_record
+        return hasattr(previous_crv, "manual_handling_record")
```

Comment:
I suggest an explicit query over `hasattr()` for checking related object existence.
`hasattr()` works for `OneToOneField` reverse relations (catches DoesNotExist), but doesn't work for `ForeignKey` (returns RelatedManager which always exists). Using `.filter(...).exists()` is more performant, consistent across relationship types, and makes the intent clearer.
For instance:
```py
return ComplianceReportVersionManualHandling.objects.filter(compliance_report_version=previous_crv).exists()
```